Makefile.envs

```diff
@@ -23,3 +23,51 @@ export PYODIDE=1
 # This is the legacy environment variable used for the aforementioned purpose
 export PYODIDE_PACKAGE_ABI=1
 export EM_COMPILER_WRAPPER=ccache
+
+export OPTFLAGS=-O2
+export CFLAGS_BASE=\
+	$(OPTFLAGS) \
+	-g \
+	-fPIC \
+	$(EXTRA_CFLAGS)
+
+export LDFLAGS_BASE=\
+	-s BINARYEN_EXTRA_PASSES="--pass-arg=max-func-params@61" \
+	$(OPTFLAGS) \
+	-s MODULARIZE=1 \
+	-s EMULATE_FUNCTION_POINTER_CASTS=1 \
+	-s LINKABLE=1 \
+	-s EXPORT_ALL=1 \
+	-s WASM=1 \
+	-std=c++14 \
+	-s LZ4=1 \
+	$(EXTRA_LDFLAGS)
+
+export SIDE_MODULE_LDFLAGS=	$(LDFLAGS_BASE) -s SIDE_MODULE=1
+export MAIN_MODULE_LDFLAGS= $(LDFLAGS_BASE) -s MAIN_MODULE=1 \
+	-s EXPORTED_FUNCTIONS='["___cxa_guard_acquire", "__ZNSt3__28ios_base4initEPv", "_main"]' \
+	$(CPYTHONROOT)/installs/python-$(PYVERSION)/lib/libpython$(PYMINOR).a \
+	-s TOTAL_MEMORY=20971520 \
+	-s ALLOW_MEMORY_GROWTH=1 \
+    --use-preload-plugins \
+	-s USE_FREETYPE=1 \
+	-s USE_LIBPNG=1 \
+	-L$(wildcard $(CPYTHONROOT)/build/sqlite*/.libs) -lsqlite3 \
+	$(wildcard $(CPYTHONROOT)/build/bzip2*/libbz2.a) \
+	-lstdc++ \
+	--memory-init-file 0 \
+
+export SIDE_MODULE_CFLAGS= $(CFLAGS_BASE)
+export MAIN_MODULE_CFLAGS= $(CFLAGS_BASE) \
+    	-Wall \
+    	-Wno-warn-absolute-paths \
+    	-Werror=unused-variable \
+    	-Werror=sometimes-uninitialized \
+    	-Werror=int-conversion \
+    	-Werror=incompatible-pointer-types \
+		-I$(PYTHONINCLUDE) 
+
+export PYTHON_CFLAGS=$(CFLAGS_BASE)
```

Comment:
Maybe keep this in `cpython/Makefile` similarly to how we keep package specific flags in their respective `meta.yaml` files?

Reply:
I guess so. I think it is unlikely that we'll want different cflags for python anyway. Will do.